create-dmg: init 1.2.2

heywoodlh opened the pull request:
## Description of changes

Adds [create-dmg](https://github.com/create-dmg/create-dmg), a DMG creator for MacOS

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

heywoodlh committed:
create-dmg: init at 1.2.2

clebs commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,31 @@
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out
+    /usr/bin/make DESTDIR=$out prefix="" install
+  '';
```
`InstallPhase` is not necessary in this case. This should yield the same result:
```
installFlags = [ "PREFIX=$(out)" ];
```

clebs reviewed (commented):
Thanks for the contribution! 🚀 
Also your commits need to be reversed (first `maintainers` and then `create-dmg init`)

heywoodlh commented:
@clebs thank you for the thoughtful review! I believe I addressed all your concerns in my most recent commit: f8ada611c660eb83d6611558081dbdbbfb50fbc8

(Sorry if the notifications I generated resolving the conversations were obnoxious)

heywoodlh commented:
Also, here's how I tested functionality (it seems to work as expected):

```
/tmp/test
❯ mkdir source_folder; echo stuff > source_folder/file.txt; echo other-stuff > source_folder/file2.txt

/tmp/test
❯ create-dmg \
        --volname "Application Installer" \
        --window-pos 200 120 \
        --window-size 800 400 \
        --icon-size 100 \
        --icon "Application.app" 200 190 \
        --hide-extension "Application.app" \
        --app-drop-link 600 185 \
        "test.dmg" \
        "source_folder/"
```

And the resulting Volume in Finder with the contents:

<img width="1470" alt="image" src="https://github.com/NixOS/nixpkgs/assets/18178614/740140c7-7e25-4502-96de-ea938bb7b847">

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,30 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "create-dmg";
+  version = "v1.2.2";
```
Version should start with a digit.

clebs replied:
When changging this, also change the revision on `fetchFromGitHub` to:
`rev = "refs/tags/v${finalAttrs.version}";`

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,30 @@
+  meta = {
+    description = "A shell script to build fancy DMGs";
+    homepage = "https://github.com/create-dmg/create-dmg";
+
```
```suggestion
```

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,30 @@
+    homepage = "https://github.com/create-dmg/create-dmg";
+
+    license = lib.licenses.mit;
+
```
```suggestion
```

heywoodlh commented:
@AndersonTorres thanks for the feedback, I believe I've addressed your feedback on the most recent commit

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,28 @@
+  src = fetchFromGitHub {
+    owner = "create-dmg";
+    repo = "create-dmg";
+    rev = "refs/tags/${finalAttrs.version}";
```
Strange. 

https://github.com/create-dmg/create-dmg/refs/tag/v1.2.2 points to nothing.

clebs replied:
It does not on github website but the tag is there: https://github.com/create-dmg/create-dmg/tags
If you clone the repo and do `git checkout refs/tags/v1.2.2` it works.

AndersonTorres replied:
So it lacks the `v` before the number.

clebs replied:
Not sure what you mean, the b is there in the checkout.
Were you maybe commenting on my other response?

AndersonTorres replied:
> Not sure what you mean, the b is there in the checkout.

It should be 

`rev = "refs/tags/v${finalAttrs.version}";`

instead of

`rev = "refs/tags/${finalAttrs.version}";`

clebs replied:
Yes, that is what o said needed to be changed in the first comment. That of the v is removed from the version attribute, it needs to be added to ref

heywoodlh replied:
Resolved on the most recent commit, please let me know if this is still an issue!

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,28 @@
+  src = fetchFromGitHub {
+    owner = "create-dmg";
+    repo = "create-dmg";
+    rev = "refs/tags/${finalAttrs.version}";
```
The intention is not overriding version per se, but not requiring rec.

AndersonTorres replied:
https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec

AndersonTorres commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,28 @@
+  src = fetchFromGitHub {
+    owner = "create-dmg";
+    repo = "create-dmg";
+    rev = "refs/tags/${finalAttrs.version}";
```
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

heywoodlh replied:
Resolved with most recent commit

heywoodlh commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation (finalAttrs: {
```
I didn't even know `stdenvNoCC` was a thing, thanks for the pointer! Switched to it on the most recent commit.

clebs reviewed (approved):
LGTM!

AndersonTorres reviewed (changes requested):
Reword the commit as 

```
create-dmg: init at 1.2.2
```

heywoodlh commented:
@AndersonTorres done

Out of curiosity and for my future reference, is the phrasing of the commit _that important_ before being merged into master? If so, why?

AndersonTorres commented:
1. This is pontificated at Contributing docs: https://github.com/NixOS/nixpkgs/blob/master/pkgs%2FREADME.md

2. The formatting allows grepping the commit logs more easily, in a similar fashion to the "conventional commits" standard (which I think it is ugly).

[XKCD 1296](https://xkcd.com/1296/)

afh commented on pkgs/by-name/cr/create-dmg/package.nix:
```diff
@@ -0,0 +1,28 @@
+  installFlags = [ "prefix=$(out)" ];
+
+  meta = {
+    description = "A shell script to build fancy DMGs";
```
Please remove the indefinite article from the beginning of the description as recommended in [section "Meta attributes" in `pkgs/README.md`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes)
```suggestion
    description = "Shell script to build fancy DMGs";
```

heywoodlh replied:
Added on most recent commit, thanks!

sarahec replied:
@afh I can confirm that change. Can we get an approval?

afh replied:
The changes look good to me, yet having all checks pass would be helpful, @sarahec.

afh reviewed (changes requested):
Thanks for this, @heywoodlh, much appreciated. Please find below a minor suggestion for improvement.

afh commented:
Something seems off with the checks, not sure if the following is going to work to re-trigger the checks…

@ofborg test

sarahec commented:
@afh the merge check times out two weeks ago. Do we ask @heywoodlh to rebase atop master and force push, knowing it'll lose the one approval (@clebs)?

AndersonTorres commented:
@ofborg build create-dmg

heywoodlh commented:
Let me know what you need from me and I'll be happy to make those changes

afh commented:
This PRs branch is ~12k commits behind `master`, possibly that is the reason why checks time out. @heywoodlh, could you rebase this PRs branch onto master, please?

DontEatOreo commented:
@heywoodlh Are you still interested in pursuing this PR?

heywoodlh commented:
Rebased onto master, let me know if there's any other follow up that is needed.

afh commented:
Thanks for rebase, @heywoodlh, mind formatting the package using:
`nix run nixpkgs#nixfmt-rfc-style -- pkgs/by-name/cr/create-dmg/package.nix`, please? This should address the failed nixfmt check..

heywoodlh commented:
@afh done! Also, added `meta.changelog` in there while I was at it.

afh commented:
Result of `nixpkgs-review pr 319628` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

afh reviewed (approved):
Thanks for the latest updates on this PR, much appreciated 👍

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1892

DontEatOreo commented:
Result of `nixpkgs-review pr 319628` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

DontEatOreo commented:
Result of `nixpkgs-review pr 319628` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>create-dmg</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1895

JohnRTitor commented:
Result of `nixpkgs-review pr 319628` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

AndersonTorres commented:
> Result of `nixpkgs-review pr 319628` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

Not conclusive, since it is Darwin-only.

JohnRTitor commented:
Yeah, tried on the nix-community darwin builder and it succeeded.
```log
johnrtitor@darwin01 ~ % nix build "github:nixos/nixpkgs/refs/pull/319628/merge#create-dmg" --print-out-paths
/nix/store/ajjyja98l3xa05w6rvrzi14arj23g4sd-create-dmg-1.2.2
nix build "github:nixos/nixpkgs/refs/pull/319628/merge#create-dmg"   1.70s user 12.60s system 60% cpu 23.516 total
johnrtitor@darwin01 ~ % ./result/bin/create-dmg
Not enough arguments. Run 'create-dmg --help' for help.
1 johnrtitor@darwin01 ~ % ./result/bin/create-dmg --help
create-dmg 1.2.1

Creates a fancy DMG file.

Usage:  create-dmg [options] <output_name.dmg> <source_folder>

All contents of <source_folder> will be copied into the disk image.

```
